src/napari_spatialdata/_utils.py

```diff
@@ -221,3 +203,272 @@ def _points_inside_triangles(points: NDArrayA, triangles: NDArrayA) -> NDArrayA:
         out[i] = _point_inside_triangles(triangles - points[i])
 
     return out
+
+
+def _transform_to_rgb(element: SpatialImage | MultiscaleSpatialImage) -> tuple[DataArray, bool]:
+    """Swap the axes to y, x, c if an image supports rgb(a) visualization.
+
+    Checks whether c dim is present in the axes and allows for rgb(a) visualization. If so, subsequently transposes it
+    into c x y x x and flags as suitable for rgb visualization.
+
+    Parameters
+    ----------
+    element: SpatialImage | MultiScaleSpatialImage
+        Element in sdata.images
+
+    Returns
+    -------
+    new_raster: DataArray
+        The image in shape of c x y x x.
+    rgb: bool
+        Flag indicating suitability for rgb(a) visualization
+    """
+    axes = get_axes_names(element)
+
+    if "c" in axes:
+        assert axes.index("c") == 0
+        if isinstance(element, SpatialImage):
+            n_channels = element.shape[0]
+        elif isinstance(element, MultiscaleSpatialImage):
+            v = element["scale0"].values()
+            assert len(v) == 1
+            n_channels = v.__iter__().__next__().shape[0]
+        else:
+            raise TypeError(f"Unsupported type for images or labels: {type(element)}")
+    else:
+        n_channels = 0
+
+    if n_channels in [3, 4]:
+        rgb = True
+        new_raster = element.transpose("y", "x", "c")
+    else:
+        rgb = False
+        new_raster = element
+
+    # TODO: after we call .transpose() on a MultiscaleSpatialImage object we get a DataTree object. We should look at
+    # this better and either cast somehow back to MultiscaleSpatialImage or fix/report this
+    if isinstance(new_raster, (MultiscaleSpatialImage, DataTree)):
+        list_of_xdata = []
+        for k in new_raster:
+            v = new_raster[k].values()
+            assert len(v) == 1
+            xdata = v.__iter__().__next__()
+            list_of_xdata.append(xdata)
+        new_raster = list_of_xdata
+
+    return new_raster, rgb
+
+
+def _init_colors_for_obs(adata: AnnData) -> AnnData:
+    from scanpy.plotting._utils import _set_colors_for_categorical_obs
```

Comment:
this we copied over it frm scanpy so to not be dependent on scanpy anymore, it's in `categorical_utils.py`